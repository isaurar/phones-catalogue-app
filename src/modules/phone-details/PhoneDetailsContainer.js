import React from 'react';
import {connect} from 'react-redux';
import * as actions from './actions';

const PhoneDetailsContainer = () => {
    return (
        <div>Phone Details Container</div>
    );
};

const mapStateToProps = ({phoneDetails}) => phoneDetails;

const mapDispatchToProps = dispatch => ({
    actions
});

export default connect(mapStateToProps, mapDispatchToProps)(PhoneDetailsContainer);