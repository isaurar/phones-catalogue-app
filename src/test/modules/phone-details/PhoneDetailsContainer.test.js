import React from 'react';
import TestRenderer from 'react-test-renderer';
import PhoneDetailsContainer from '../../../modules/phone-details/PhoneDetailsContainer.js';
import { MemoryRouter } from 'react-router-dom';

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn()
    .mockReturnValue({}),
  useDispatch: jest.fn()
}));

describe('phone details container', () => {
  test('should be rendered phone details components correctly', () => {
    const rendered = TestRenderer.create(
      <MemoryRouter>
        <PhoneDetailsContainer />
      </MemoryRouter>
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
}); 