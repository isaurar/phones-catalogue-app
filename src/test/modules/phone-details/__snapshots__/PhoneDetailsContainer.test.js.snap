// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phone details container should be rendered phone details components correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Back
        </span>
      </button>
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <img
      alt={
        Array [
          Array [
            "brand",
            undefined,
          ],
          Array [
            "technology",
            undefined,
          ],
          Array [
            "announced",
            undefined,
          ],
          Array [
            "dimensions",
            undefined,
          ],
        ]
      }
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="600"
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      />
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        BRAND
        : 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        TECHNOLOGY
        : 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        ANNOUNCED
        : 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        DIMENSIONS
        : 
      </p>
    </div>
  </div>
</div>
`;
